firehose-source/README.md

```diff
@@ -0,0 +1,43 @@
+Spring Cloud Stream Firehose Sample
+=============================
+
+In this *Spring Cloud Stream* sample, a source emits events from a [doppler](https://github.com/cloudfoundry/loggregator) endpoint
+
+## Requirements
+
+To run this sample, you will need to have installed:
+
+* Java 7 or Above
+* A Redis server
+* [Cloudfoundry](https://github.com/cloudfoundry/cf-release), [Lattice](http://lattice.cf) or a doppler [simulator](https://github.com/viniciusccarvalho/doppler-simulator) running
+* A [protobuf](https://developers.google.com/protocol-buffers) compiler installed and available on your path
+
+## Code Tour
+
+This sample connects to a doppler remote websocket endpoint and emits events from it. The sample converts the binary protocol buffer message into either a XD Tuple or a JSON String, depending on your configuration
```

Comment:
avoid "XD" here... the Tuple class will likely move to a lower level project (e.g. spring-integration) or a "commons" repo (e.g. data-commons)